src/psyclone/tests/gocean/transformations/globalstoargs_test.py

```diff
@@ -0,0 +1,260 @@
+    ''' Check the KernelGlobalsToArguments transformation '''
+    from psyclone.transformations import KernelGlobalsToArguments
+    from psyclone.psyir.backend.fortran import FortranWriter
+
+    trans = KernelGlobalsToArguments()
+    assert trans.name == "KernelGlobalsToArguments"
+    assert str(trans) == "Convert the global variables used inside the " \
+        "kernel into arguments and modify the InvokeSchedule to pass them" \
+        " in the kernel call."
+
+    # Construct a testing InvokeSchedule
+
+    _, invoke_info = parse(os.path.join(BASEPATH, "gocean1p0",
+                                        "single_invoke_kern_with_use.f90"),
+                           api=API)
+    psy = PSyFactory(API).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    kernel = invoke.schedule.coded_kernels()[0]
+
+    # Monckeypatch resolve_deferred to avoid module searching and importing
+    # in this test. In this case we assume it is a constant INTEGER
+    for var in kernel.get_kernel_schedule().symbol_table.global_datasymbols:
+        def set_to_real(variable):
+            variable._datatype = DataType.INTEGER
+            variable.constant_value = 1
+        monkeypatch.setattr(var, "resolve_deferred", lambda: set_to_real(var))
+
+    # Test transforming a single kernel
+    trans.apply(kernel)
+
+    fwriter = FortranWriter()
+    kernel_code = fwriter(kernel.get_kernel_schedule())
+
+    assert "subroutine kernel_with_use_code(ji,jj,istep,ssha,tmask,rdt)" \
+        in kernel_code
+    assert "integer, intent(in) :: rdt" in kernel_code
+
+
+def test_globalstoarguments_multiple_kernels():
+    ''' Check the KernelGlobalsToArguments transformation with an invoke with
+    three kernel calls, two of them duplicated and the third one sharing the
+    same imported module'''
+    from psyclone.transformations import KernelGlobalsToArguments
+    from psyclone.psyir.backend.fortran import FortranWriter
+    fwriter = FortranWriter()
+
+    # Construct a testing InvokeSchedule
+    _, invoke_info = parse(os.path.
+                           join(BASEPATH, "gocean1p0",
+                                "single_invoke_three_kernels_with_use.f90"),
+                           api=API)
+    psy = PSyFactory(API).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    trans = KernelGlobalsToArguments()
+
+    expected = [
+        ["subroutine kernel_with_use_code(ji,jj,istep,ssha,tmask,rdt)",
+         "real, intent(inout) :: rdt"],
+        ["subroutine kernel_with_use2_code(ji,jj,istep,ssha,tmask,cbfr,rdt)",
+         "real, intent(inout) :: cbfr\n  real, intent(inout) :: rdt"],
+        ["subroutine kernel_with_use_code(ji,jj,istep,ssha,tmask,rdt)",
+         "real, intent(inout) :: rdt"]]
+
+    for num, kernel in enumerate(invoke.schedule.coded_kernels()):
+        kschedule = kernel.get_kernel_schedule()
+
+        # We already tested that DEFERRED globals are resolved in last test,
```

Comment:
This comment isn't quite true any more. It's tested somewhere but not in the last test.

Reply:
Remove this part of the sentence